### PR TITLE
Kill infomanager global

### PR DESCRIPTION
**Pull request by ksooo**
@garbear @FernetMenta -1 global  ;-)

EDIT: commits 3 to 5 are slightly unrelated and just small fixes for the recent guiinfo manager refactoring.

### CODE REVIEW

File: xbmc/guilib/GUIControlFactory.h [hunk 1]
```diff
@@ -35,10 +35,20 @@
 
 class CTextureInfo; // forward
 class CAspectRatio;
-class CGUIInfoLabel;
 class TiXmlNode;
 class CGUIAction;
 
+namespace KODI
```
[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/guilib/GUIControlFactory.h]: > can we stop using namespace KODI? does not make much sense

Well, there are people in the team having a different opinion on this. For my last PR I was asked by a reviewer to introduce this namespace. That's no fun.

[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/guilib/GUIControlFactory.h]: So, as long as there is no team agreement on this I would like to keep this as it is. Once we have a discision we can change the code base to follow that rule.

[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/guilib/GUIControlFactory.h]: Or do I mistunderstand you and you mean the problem is to put classes directly in the namespace KODI? If so, I don't do this. I put everything in namespace KODI::GUILIB::GUIINFO - aligned with the directory structure of the affected source files xbmc/guilib/guiinfo...

[the remark this answers is not in the document]
Comment (FernetMenta) [reply xbmc/guilib/GUIControlFactory.h]: nevermind :)

File: xbmc/filesystem/LibraryDirectory.cpp [hunk 2]
```diff
@@ -152,7 +152,7 @@ TiXmlElement *CLibraryDirectory::LoadXML(const std::string &xmlFile)
 
   // check the condition
   std::string condition = XMLUtils::GetAttribute(xml, "visible");
-  if (condition.empty() || g_infoManager.EvaluateBool(condition))
+  if (condition.empty() || CServiceBroker::GetGUI()->GetInfoManager().EvaluateBool(condition))
```
Comment (FernetMenta) [inline xbmc/filesystem/LibraryDirectory.cpp]: this should check for nullptr. ideally filesystem would not call into guilib at all

File: xbmc/music/MusicDatabase.cpp [hunk 3]
```diff
@@ -6407,7 +6407,7 @@ bool CMusicDatabase::CommitTransaction()
 {
   if (CDatabase::CommitTransaction())
   { // number of items in the db has likely changed, so reset the infomanager cache
-    g_infoManager.GetInfoProviders().GetLibraryInfoProvider().SetLibraryBool(LIBRARY_HAS_MUSIC, GetSongsCount() > 0);
+    CServiceBroker::GetGUI()->GetInfoManager().GetInfoProviders().GetLibraryInfoProvider().SetLibraryBool(LIBRARY_HAS_MUSIC, GetSongsCount() > 0);
```
Comment (FernetMenta) [inline xbmc/music/MusicDatabase.cpp]: better check for nullptr

File: xbmc/filesystem/LibraryDirectory.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/filesystem/LibraryDirectory.cpp]: okay, will add the check.

File: xbmc/music/MusicDatabase.cpp (hunk 3 above)
Comment (ksooo) [reply xbmc/music/MusicDatabase.cpp]: okay.

File: xbmc/profiles/ProfilesManager.cpp [hunk 4]
```diff
@@ -281,9 +281,10 @@ bool CProfilesManager::LoadProfile(size_t index)
 
   CServiceBroker::GetInputManager().SetMouseEnabled(CServiceBroker::GetSettings().GetBool(CSettings::SETTING_INPUT_ENABLEMOUSE));
 
-  g_infoManager.ResetCache();
-  g_infoManager.GetInfoProviders().GetGUIControlsInfoProvider().ResetContainerMovingCache();
-  g_infoManager.GetInfoProviders().GetLibraryInfoProvider().ResetLibraryBools();
+  CGUIInfoManager& infoMgr = CServiceBroker::GetGUI()->GetInfoManager();
```
Comment (FernetMenta) [inline xbmc/profiles/ProfilesManager.cpp]: this should check for nullptr, too. iirc profiles get loaded before ui

Comment (ksooo) [reply xbmc/profiles/ProfilesManager.cpp]: okay.

**Review by FernetMenta — APPROVED** [review]
there are just a few locations outside gui where checking for nullptr would make sense. If you got time, would be great to do this. Otherwise we can fix it, if it crashed somewhere :)

**Pull request merged**